src/compiler/scala/tools/nsc/javac/JavaParsers.scala

```diff
@@ -726,8 +726,15 @@ trait JavaParsers extends ast.parser.ParsersCommon with JavaScanners {
           in.nextToken()
         } else {
           if (in.token == ENUM || definesInterface(in.token)) mods |= Flags.STATIC
-          val decls = memberDecl(mods, parentToken)
-          (if (mods.hasStaticFlag || inInterface && !(decls exists (_.isInstanceOf[DefDef])))
+          val decls = joinComment(memberDecl(mods, parentToken))
+
+          def isDefDef(tree: Tree): Boolean = tree match {
+            case _: DefDef => true
+            case DocDef(_, defn) => isDefDef(defn)
+            case _ => false
+          }
+
+          (if (mods.hasStaticFlag || inInterface && !(decls exists isDefDef))
```

Comment:
`DocDef`-s have a long history of such bugs :(

e.g. 1c6969945f209dafbe3c3fb38f1c3f9a5aaadb82 / c4561c1d4945a38febc41436ed333569d0e9a063

If you're motivated to do so, I'd love to see a followup change that changes the parser to no longer emit `DocDef` nodes at all, but instead parse the regular tree with an attachment `tree.updateAttachment(DocComment(...))`.